### PR TITLE
Add Script::load function

### PR DESCRIPTION
**Pull request by zhiburt**
Hi there

This PR exposes a load function to not do it manually.
It also changes a repeat logic in sync version of `Script::invoke` to limit the repeat cycle to 2 attempts (The same way as is done in async version).

Take care.

### CODE REVIEW

**Comment by zhiburt** [comment]
Sorry.....
one more time, please?

*It must be it

File: src/script.rs [hunk 1]
```diff
@@ -124,26 +124,26 @@ impl<'a> ScriptInvocation<'a> {
...
-                Err(err) => {
-                    if err.kind() == ErrorKind::NoScriptError {
-                        cmd("SCRIPT")
-                            .arg("LOAD")
-                            .arg(self.script.code.as_bytes())
-                            .query(con)?;
-                    } else {
-                        fail!(err);
-                    }
+        let mut eval_cmd = cmd("EVALSHA");
```
Comment (djc) [inline src/script.rs]: Why do you think we no longer need the `loop`?

File: src/script.rs [hunk 2]
```diff
@@ -204,13 +182,25 @@ impl<'a> ScriptInvocation<'a> {
...
-            .query_async(con).await?;
+        let hash: String = self.load_cmd().query_async(con).await?;
 
         debug_assert_eq!(hash, self.script.hash);
 
         Ok(hash)
     }
+
+    fn load_cmd(&self) -> Cmd {
+        let mut cmd = cmd("SCRIPT");
```
Comment (djc) [inline src/script.rs:193]: Style nit: I think we could just chain this directly all the way through instead of introducing a variable binding?

File: src/script.rs [hunk 3]
```diff
@@ -204,13 +182,25 @@ impl<'a> ScriptInvocation<'a> {
...
+        cmd.arg("LOAD").arg(self.script.code.as_bytes());
+        cmd
+    }
+
+    fn eval_cmd(&self) -> Cmd {
+        let mut cmd = cmd("EVALSHA");
```
Comment (djc) [inline src/script.rs:199]: Same here.

File: tests/test_async_async_std.rs [hunk 4]
```diff
@@ -295,14 +295,11 @@ fn test_script_load() {
     let script = redis::Script::new("return 'Hello World'");
 
     block_on_all(async move {
-        let mut con = ctx.multiplexed_async_connection_async_std().await?;
-
-        let hash = script.prepare_invoke().load_async(&mut con);
-        assert_eq!(hash, Ok(script.get_hash().to_string()));
+        let mut con = ctx.multiplexed_async_connection_async_std().await.unwrap();
```
Comment (djc) [inline tests/test_async_async_std.rs:298]: Why did you change these to `unwrap()`?

**Review by djc — COMMENTED** [review]
Thanks, this mostly looks good!

File: src/script.rs (hunk 1 above)
Comment (zhiburt) [reply src/script.rs]: As you may see that a sync and async implementation is different.
I think that the async implementation with 1 retry is better. So I replace a loop implementation with it.

The loop may cause a number of retries or even "potentially" an infinite loop.
See when there's no script `LOAD`ed we load it, but then if someone `FLUSH`es redis `EVAL` will be an error an we will try to `LOAD` the script one more time.

Also it's beneficial because the unification of the code.

File: tests/test_async_async_std.rs (hunk 4 above)
Comment (zhiburt) [reply tests/test_async_async_std.rs:298]: To fix the build :)

Do you think it matters?

File: src/script.rs (hunk 2 above)
Comment (djc) [reply src/script.rs:193]: Fair enough.

**Review by djc — APPROVED** [review]
